libs/deposits/src/lib/deposit-form.tsx

```diff
@@ -154,16 +147,30 @@ export const DepositForm = ({
         )}
       </FormGroup>
       <FormGroup label={t('Asset')} labelFor="asset" className="relative">
-        <Select {...register('asset', { validate: { required } })} id="asset">
-          <option value="">{t('Please select')}</option>
-          {assets
-            .filter((a) => a.source.__typename === 'ERC20')
-            .map((a) => (
-              <option key={a.id} value={a.id}>
-                {a.name}
-              </option>
-            ))}
-        </Select>
+        <Controller
+          control={control}
+          name="asset"
+          rules={{ validate: { required } }}
+          render={({ field }) => (
+            <Select
+              id="asset"
+              {...field}
+              onChange={(e) => {
+                field.onChange(e);
+                onSelectAsset(e.target.value);
+              }}
+            >
+              <option value="">{t('Please select')}</option>
+              {assets
+                .filter((a) => a.source.__typename === 'ERC20')
```

Comment:
I don't like this 'ERC20' literal. It's repeated couple times across project
I think we should have const and use it instead
```
export const ERC20 = 'ERC20';
export interface ERC20AssetSource {
  __typename: typeof ERC20;
  contractAddress: string;
}
if (asset?.source.__typename === ERC20) ...
```

Reply:
Moved types and added a isAssetTypeERC20 function to libs/react-helpers/assets. Saves some duplication too